src/nvim/screen.c

```diff
@@ -4542,12 +4556,29 @@ void status_redraw_all(void)
   }
 }
 
+// Mark all window bars for redraw.
+void winbar_redraw_all(void)
+{
+  FOR_ALL_WINDOWS_IN_TAB(wp, curtab) {
+    if (wp->w_winbar_height) {
+      wp->w_redr_winbar = true;
+      redraw_later(wp, VALID);
+    }
+  }
+}
+
```

Comment:
let's not mindlessly repeat all code for statusline. if we need to redraw _all_ statuslines, we also need to redraw all winbars. This can be handled in one single function.

Reply:
Done. Take another look and tell me if it's good now